### PR TITLE
Fixes #RHINENG-765 - fixed incorrectly generating url

### PR DESCRIPTION
**Pull request by patilsuraj767**
fixes - https://github.com/RedHatInsights/ansible-collections-insights/issues/48 
This can be improved more but here is a quick fix. Basically URL is generating incorrectly. We are appending to the same old url because of which url gets bigger than 4096 characters.
Example - 
```
https://console.redhat.com/api/inventory/v1/hosts?&page=2&page=3&page=4&page=5&page=6&page=7&page=8&page=9&page=10&page=11
```

### CODE REVIEW

**Review by kruai — APPROVED** [review]
Definitely an improvement; this should make it so it doesn't keep appending to itself ad infinitum.

File: plugins/inventory/insights.py
Comment (kruai) [inline plugins/inventory/insights.py:249]: Actually, this one should probably be:
```suggestion
                        hosts_url = "%s&tags=%s" % (hosts_url, '&tags='.join(filter_tags))
```
Otherwise, the `&page=` params above would get entirely overwritten.

**Review by kruai — CHANGES_REQUESTED** [review]
Missed something in my last review

**Comment by amarmh** [comment]
Tested in local setup:

Code:  ansible/collections/ansible_collections/redhat/insights/plugins/inventory/insights.py
~~~
233         #while url:
234         #    response = self.session.get(url, auth=self.auth, headers=self.headers)
235         hosts_url = url
236         while hosts_url:
237             print("=========================================================")
238             print(hosts_url)
239             response = self.session.get(hosts_url, auth=self.auth, headers=self.headers)
240 
241             if response.status_code != 200:
242                 raise AnsibleError("http error (%s): %s" %
243                                    (response.status_code, response.text))
244             elif response.status_code == 200:
245                 results += response.json()['results']
246                 total = response.json()['total']
247                 count = response.json()['count']
248                 per_page = response.json()['per_page']
249                 page = response.json()['page']
250                 if per_page * (page - 1) + count < total:
251                     hosts_url = "%s&page=%s" % (url, (page + 1))
252                     #url = "%s&page=%s" % (url, (page + 1))
253                     if len(filter_tags) > 0:
254                         hosts_url = "%s&tags=%s" % (hosts_url, '&tags='.join(filter_tags))
255                         #url = "%s&tags=%s" % (url, '&tags='.join(filter_tags))
256                 else:
257                     #url = None
258                     hosts_url = None
259 
~~~

Output on Controller UI:
~~~
ansible-inventory [core 2.14.6]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/runner/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.9/site-packages/ansible
  ansible collection location = /home/runner/.ansible/collections:/usr/share/ansible/collections:/usr/share/automation-controller/collections
  executable location = /usr/bin/ansible-inventory
  python version = 3.9.16 (main, May 31 2023, 12:21:58) [GCC 8.5.0 20210514 (Red Hat 8.5.0-18)] (/usr/bin/python3.9)
  jinja version = 3.1.2
  libyaml = True
Using /etc/ansible/ansible.cfg as config file
setting up inventory plugins
Loading collection redhat.insights from /usr/share/ansible/collections/ansible_collections/redhat/insights
[WARNING]: Collection redhat.insights does not support Ansible version 2.14.6
Using inventory plugin 'ansible_collections.redhat.insights.plugins.inventory.insights' to process inventory source '/runner/inventory/insights.yml'
=========================================================
https://cloud.redhat.com/api/inventory/v1/hosts?&staleness=fresh&staleness=stale&staleness=unknown&registered_with=insights
=========================================================
https://cloud.redhat.com/api/inventory/v1/hosts?&staleness=fresh&staleness=stale&staleness=unknown&registered_with=insights&page=2
=========================================================
https://cloud.redhat.com/api/inventory/v1/hosts?&staleness=fresh&staleness=stale&staleness=unknown&registered_with=insights&page=3
=========================================================
https://cloud.redhat.com/api/inventory/v1/hosts?&staleness=fresh&staleness=stale&staleness=unknown&registered_with=insights&page=4
=========================================================
https://cloud.redhat.com/api/inventory/v1/hosts?&staleness=fresh&staleness=stale&staleness=unknown&registered_with=insights&page=5
=========================================================
https://cloud.redhat.com/api/inventory/v1/hosts?&staleness=fresh&staleness=stale&staleness=unknown&registered_with=insights&page=6
~~~

**Comment by kruai** [comment]
@patilsuraj767 Is this waiting on reviews from anyone else? Also, should all of the PR checks be failing...?

**Comment by patilsuraj767** [comment]
> @patilsuraj767 Is this waiting on reviews from anyone else? Also, should all of the PR checks be failing...?

@kruai I am not sure who is the maintainer of the repo and who has the merge rights. 
I can see highest commits are from @willtome and @SteveHNH may be they can help with PR checks and help to get the PR merged.

**Pull request merged**
